newroutes/auth.js

```diff
@@ -0,0 +1,99 @@
+const autoBind = require("auto-bind")
+const express = require("express")
+
+// Import middleware
+const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+
+const { FRONTEND_URL } = process.env
+
+const AUTH_TOKEN_EXPIRY_MS = parseInt(
+  process.env.AUTH_TOKEN_EXPIRY_DURATION_IN_MILLISECONDS,
+  10
+)
+const CSRF_TOKEN_EXPIRY_MS = 600000
+const CSRF_COOKIE_NAME = "isomer-csrf"
+const COOKIE_NAME = "isomercms"
+
+class AuthRouter {
+  constructor({ authService }) {
+    this.authService = authService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  async authRedirect(req, res) {
+    const {
+      redirectUrl,
+      cookieToken,
+    } = await this.authService.getAuthRedirectDetails()
+    const csrfTokenExpiry = new Date()
+    csrfTokenExpiry.setTime(csrfTokenExpiry.getTime() + CSRF_TOKEN_EXPIRY_MS)
+    const cookieSettings = {
+      expires: csrfTokenExpiry,
+      httpOnly: true,
+      secure:
+        process.env.NODE_ENV !== "DEV" &&
+        process.env.NODE_ENV !== "LOCAL_DEV" &&
+        process.env.NODE_ENV !== "test",
```

Comment:
if this pattern is going to be repeated alot, we should probably have a helper to keep them in sync

Reply:
https://github.com/isomerpages/isomercms-backend/pull/328/commits/370e105d57b130be7ecda03850e5280a91fc4c70